### PR TITLE
Feat: update header to v2

### PR DESCRIPTION
**Pull request by lolottetheclash**
Thank you for reading this [issue](https://github.com/okp4/ui/issues/562) in order to have all the details of this PR.
Some screenshots:

Capture d’écran 2022-10-10 à 11 59 59

Capture d’écran 2022-10-10 à 11 56 31

Capture d’écran 2022-10-10 à 12 00 11

Capture d’écran 2022-10-10 à 12 00 22

### CODE REVIEW

File: src/ui/molecules/header/Header.tsx [hunk 1]
```diff
@@ -32,18 +51,110 @@ const BurgerMenu = ({
...
+}: DeepReadonly<{
+  onSubItemSelection: () => void
+  subMenu: JSX.Element[]
+}>): JSX.Element => {
+  return (
+    <div className="okp4-header-navigation-submenu">
+      {subMenu.map((navItem: DeepReadonly<JSX.Element>, index: number) => (
+        <div
+          className="okp4-header-navigation-submenu-item"
+          key={index}
```
Comment (RaphaelEscrig) [inline src/ui/molecules/header/Header.tsx:82]: Here, the use of the index as a key can cause issues [more info here](https://reactjs.org/docs/lists-and-keys.html).

**Review by RaphaelEscrig — CHANGES_REQUESTED** [review]
Really nice way to add easily the submenus

File: src/ui/molecules/header/Header.tsx (hunk 1 above)
Comment (lolottetheclash) [reply src/ui/molecules/header/Header.tsx:82]: Yes I know, but I really don't see how to handle keys based on jsx elements that don't have ids or other values that I can use....

Comment (RaphaelEscrig) [reply src/ui/molecules/header/Header.tsx:82]: Maybe you can use the `short.generate()` function as it's already imported in the file

Comment (lolottetheclash) [reply src/ui/molecules/header/Header.tsx:82]: i thought about it, but I think it's useless... check the answer of this : https://stackoverflow.com/questions/52196127/different-ways-to-add-a-key-to-jsx-element-in-loop-in-react

**Review by ErikssonJoakim — APPROVED** [review]
Nice work looks good to me 👍

**Review by RaphaelEscrig — APPROVED** [review]

**Review by JoeyDjr — APPROVED** [review]
LGTM

File: src/ui/molecules/header/Header.tsx [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+import type { DeepReadonly, UseState } from 'superTypes'
 import { useMediaType } from 'hook/useMediaType'
+import { useOnKeyboard } from 'hook/useOnKeyboard'
+import { useOnClickOutside } from 'hook/useOnClickOutside'
 import { ThemeSwitcher as Switcher } from 'ui/atoms/theme/ThemeSwitcher'
+import { Typography } from 'ui/atoms/typography/Typography'
 import { Icon } from 'ui/atoms/icon/Icon'
 import './header.scss'
 
+type NavigationItemsWithId = DeepReadonly<Map<string, NavigationItem>>
```
Comment (fredericvilcot) [inline src/ui/molecules/header/Header.tsx:13]: As the `Map` keys are ids, I will have named it simply `NavigationItems`

File: src/ui/molecules/header/Header.tsx [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+
+export type NavigationMenuProps = {
+  navigation: NavigationItemsWithId
+  onNavItemSelection: (id: string, isSubItem?: boolean) => void
```
Comment (fredericvilcot) [inline src/ui/molecules/header/Header.tsx:17]: It will be better to create an alias for id
```ts
type Id = string
```

**Review by fredericvilcot — COMMENTED** [review]
Great, just minor remarks ...

File: src/ui/molecules/header/Header.tsx (hunk 3 above)
Comment (lolottetheclash) [reply src/ui/molecules/header/Header.tsx:17]: Good idea!

**Review by fredericvilcot — APPROVED** [review]
LGTM!

**Review by ccamel — APPROVED** [review]

**Pull request merged**